compiler/semantic/analyzer.go

```diff
@@ -14,28 +14,114 @@ import (
 // to DAG form, resolving syntax ambiguities, and performing constant propagation.
 // After semantic analysis, the DAG is ready for either optimization or compilation.
 func Analyze(ctx context.Context, seq ast.Seq, source *data.Source, head *lakeparse.Commitish) (dag.Seq, error) {
-	c := newAnalyzer(ctx, source, head)
-	return c.check(seq)
+	a := newAnalyzer(ctx, source, head)
+	s, err := a.check(seq)
```

Comment:
Maybe just call a.semSeq directly?
```suggestion
	s, err := a.semSeq(seq)
```

Reply:
Yeah...